### DISCUSSION TITLE
posts/a-nu-alternative-to-the-old-makefile/

### DISCUSSION THREAD

**Discussion by giscus** [amtoine/amtoine.github.io#6] [blog]
# posts/a-nu-alternative-to-the-old-makefile/

How to setup a Nushell-based alternative to `make`

https://amtoine.github.io/posts/a-nu-alternative-to-the-old-makefile/

**Comment by kai-tub** [comment]
Really interesting blog post!
I've been thinking about a similar application but one thing I am missing is that the current solution requires you to activate nushell first and then to execute the module/hook.
For other users that do not want to switch to nu (yet) might have no issue with installing it as a dependency of the project but would still prefer to use the shell of their choice and simple execute `nu-make` which would be a nu script wrapper around the main entrypoint and exposes all the possible commands + arguments. :)

Keep on blogging!

**Reply by amtoine** [reply]
> Really interesting blog post!
>
glad you liked it :blush: 

> I've been thinking about a similar application but one thing I am missing is that the current solution requires you to activate nushell first and then to execute the module/hook.
>
yes this is true...
this is why i have written a little *hook* to do that for me :yum: 

> For other users that do not want to switch to nu (yet) might have no issue with installing it as a dependency of the project but would still prefer to use the shell of their choice and simple execute `nu-make` which would be a nu script wrapper around the main entrypoint and exposes all the possible commands + arguments. :)
> 
this is definitely a limitation of this approach :confused: 
my goal here really is to leverage the power of Nushell as much as possible :smirk: 
i think modules are more powerful in that regard than scripts, which is why i write my toolkits as modules and not as scripts :open_mouth: 

> Keep on blogging!
>
will do! thanks for the kind words :sparkles:

**Comment by xie-zheng** [comment]
The hook part is illuminating, i used to manually search for project files to assert the current project env in my custom `build` function.

**Reply by amtoine** [reply]
glad it comes in handy :yum: 

nowadays i use a simpler single hook to load the toolkit as an overlay, mainly to rename the command to `tk`, but the idea stays the same:
```nushell
{
    condition: {|_, after| $after | path join 'toolkit.nu' | path exists }
    code: "overlay use --prefix toolkit.nu as tk"
}
```
